src/_P088_Ping.ino

```diff
@@ -111,67 +177,22 @@ boolean Plugin_088(byte function, struct EventStruct *event, String& string)
 
   case PLUGIN_EXIT:
   {
+    clearPluginTaskData(event->TaskIndex);
     P088_data->instances--;
```

Comment:
P088_data can be a nullptr.

Reply:
Does it mean we can have PLUGIN_EXIT calls more than PLUGIN_INIT, or PLUGIN_EXIT without PLUGIN_INIT?
This structure allocated if at least one ping plugin exist, and deallocate on last one EXIT

Reply:
If a plugin was added, but never enabled, then there is no call to the init function.
So everything that should have been done in the init function is not guaranteed to  have happened if you remove the plugin.

Reply:
Ok

Reply:
Then it complicate things a little bit. P088_data should be allocated (and PCB within, shared by all instances) should be active only if at least one of plugin instances is active.
I guess best place to create/destroy it - inside constructor/destructor of PluginTaskData.
I commited new code, can you check it please if it is ok?